recipes-mender/mender/files/mender-restore-saved-settings.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+#
+
+SAVED_SETTINGS_FILE=/data/mender/hosted-mender-settings.dat
+MENDER_CONF_FILE=/etc/mender/mender.conf
+
+if [ -e "${SAVED_SETTINGS_FILE}" ]; then
```

Comment:
Maybe it would be cleaner if we added this to the python script's main as a test before launching the GUI application?

Reply:
For example
```
if __name__ == "__main__":
    if os.getcwd() != "/etc/mender/login":
        # Make sure we're in the correct directory
        if not os.path.exists("/etc/mender/login"):
            os.makedirs("/etc/mender/login")
        os.chdir("/etc/mender/login")
    if os.path.exists("/data/mender/hosted-mender-settings.dat"):
        with open("/data/mender/hosted-mender-settings.dat") as f_hosted:
            with open("/etc/mender/mender.conf") as f_mender:
                conf_mender = json.load(f_mender, "w+")
                # if json is used to store the serverurl and tenant token
                conf_hosted = json.load(f_hosted, "r")
                if "ServerURL" in conf_hosted.keys():
                    conf_mender["ServerURL"] = conf_mender["ServerURL"]
                if "TenantToken" in conf_hosted.keys():
                    conf_mender["TenantToken"] = conf_mender["TenantToken"]
                f_mender.seek(0,0)
                f_mender.truncate(0)
                json.dump(conf_mender, f_mender)
        sys.exit(0)
    # Start Qt widget
    app = QtWidgets.QApplication(sys.argv)
    # Create log in window
    win = LogInWindow()
    # Display window
    win.show()
    # Execute widget application threads
    app.exec_()
```

Reply:
I was worried about a race condition with the systemd startup script.  This way it's guaranteed to be completed _before_ the mender client launches.